scss/boost_union/post.scss

```
@@ -21,11 +21,17 @@
     }
     /* Make sure that the back to top button is not covered by anything else. */
     z-index: $zindex-dropdown;
+    /* Animate position together with footer button. */
+    transition: right 0.2s;
 }
 #back-to-top i::before {
     /* Move the caret icon slightly up for a nicer look. */
     vertical-align: 0.3rem;
 }
+#page.drawers.show-drawer-right #back-to-top {
+    /* Move the back to top button when right drawer is shown. */
+    right: calc(315px + 2rem);
```

Looking at the place where Moodle core sets this style for the footer button on https://github.com/moodle/moodle/blob/master/theme/boost/scss/moodle/layout.scss#L212-L214, you will see that these 315px come from a SCSS variable.

So, this line should say:

```suggestion
    right: calc(#{$drawer-right-width} + 2rem);
```

PS: It's not necessary to add the .jsenabled class as well as the back-to-top button is not shown at all without JS enabled.